sherpa/astro/data.py

```diff
@@ -3427,7 +3427,7 @@ def get_filter(self, group=True, format='%.12f', delim=':'):
 
         # Ensure the data is in ascending order for create_expr.
         #
-        if self.units == 'wavelength':
+        if len(x) > 0 and x[-1] < x[0]:
```

Comment:
I was wondering about a len check. I **think** at this point we're guaranteed that x and mask are not empty, but it doesn't hurt to add the check here.